Change lookup_class_name method to always return string.

Hamdiakoguz opened the pull request:
When profiling code with a class which redefines 'name' class method to return a
symbol instead of default name, top_n method fails with error:
"... memory_profiler/top_n.rb:13:in `sort_by': comparison of Array with Array
failed (ArgumentError)".

So I changed lookup_class_name to call to_s on klass.name.

SamSaffron commented:
hmmm what if name is `nil` you will no longer get "<<unknown>>" ?

dgynn commented:
We could move the `.to_s` outside to handle `nil` so it would be...
```ruby
@class_name_cache[klass] ||= ((klass.is_a?(Class) && klass.name) || '<<Unknown>>').to_s
```

@Hamdiakoguz, I'm all for making this more bulletproof and thanks for adding the test case. Do you know what is creating classes like this? Is it a mocking framework?

Hamdiakoguz commented:
@SamSaffron I updated the code as proposed by @dgynn.

@dgynn It was an internal library of ours that created those classes with overriden names. 

Do you know why tests are failing on travis? Ruby 2.2.4 test passed after I updated code but now ruby head failed. 
Also I run tests with ruby 2.1.0 on my machine with no problem. But when I try to run them with 2.2.4 test_reporter tests are failing.

dgynn commented:
It looks like ruby-head is flaky right now. I see the same errors on master and I see stack dumps when I try to debug it. I've excluded it from travis for failures.

Thanks for the patch, @Hamdiakoguz.
